hoomd/HOOMDVersion.h.in

```diff
@@ -7,10 +7,6 @@
 
 // Full text HOOMD version string
 #define HOOMD_VERSION "${HOOMD_VERSION}"
-// Full text HOOMD version string
-#define HOOMD_VERSION_LONG "${HOOMD_VERSION_LONG}"
-// Text string containing the date this build was configured
-#define COMPILE_DATE "${COMPILE_DATE}"
```

Comment:
Why are we moving this from C++ to Python. I don't have a preference necessarily, but wondering why the change was made.

Reply:
We don't use these in C++ anywhere important so they don't need to be there. Since `COMPILE_DATE` changes every day and `HOOMD_VERSION_LONG` includes the git SHA, these cause the header to change regularly and make for a number of needless recompiles and relinks. Since I am refactoring this file anyways, I thought to solve that issue. It does cause some difficulty for Sphinx docs, though so maybe I will just revert the change.